### DISCUSSION TITLE
How does one access state from a store in another store?

### DISCUSSION THREAD

**Discussion by haibert** [pmndrs/zustand#630] [Q&A]
in Redux its done like so for example:  const userID = getState().userReducer.userInfo.userID. 
You can access the state of another reducer with this line, what would be the zustand equivalent of this ?

**Comment by dai-shi** [comment]
```js
const useStore = create(() => ({
  count: 0,
}))

console.log(useStore.getState().count)
```
Is it what you are asking?

**Reply by haibert** [reply]
No something like this. 
```
const useStore = create(() => ({ 
 count: 0, 
})) 
```

```
const useStoreTwo = create(() => ({ 
 count2: 0, 
 someFunction: () => {
 I want to access state from useStore here in this function inside this other separate store 
}, }))
```

**Reply by dai-shi** [reply]
This still works. Note that `getState()` is not reactive.

```js
const useStore = create(() => ({ 
  count: 0, 
})) 

const useStoreTwo = create(() => ({ 
  count2: 0, 
  someFunction: () => {
    const count = useStore.getState().count
    // ...
  },
}))
```

**Reply by haibert** [reply]
Thank you! Im sorry but what do you mean by not reactive could you elaborate on that? And this is not an anti pattern right? It wont cause any performance issues or such ?

**Reply by dai-shi** [reply]
By not reactive, I mean, when the `count` value changes in the first store, the `someFunction` is not automatically re-evaluated. If that's expected, it's not an anti pattern and no performance issue. That said, if two stores are not isolated, my first choice would be a combined single store. fyi: https://github.com/pmndrs/zustand/wiki/Flux-inspired-practice

**Reply by haibert** [reply]
Thank you very much for the quick response. Coming from Redux... I must tell you that this is by far the best state management library. It is so simple and such a delight to work with. You guys did a fantastic job. Thank you for the hard work.

**Reply by napa3um** [reply]
https://github.com/pmndrs/zustand/issues/44

**Reply by Nishchit14** [reply]
> This still works. Note that `getState()` is not reactive.
> 
> ```js
> const useStore = create(() => ({ 
>   count: 0, 
> })) 
> 
> const useStoreTwo = create(() => ({ 
>   count2: 0, 
>   someFunction: () => {
>     const count = useStore.getState().count
>     // ...
>   },
> }))
> ```

I am trying to call an action of first `store` from the action of `storeTwo`, here both stores are created from `zustand/context`

Here is the context API
```js
const { Provider,   useStore , useStoreApi } = createContext();
```

and calling like this
/store/store.two.js 
```js
// this is a separate file, non react environment

const useStoreTwo = create(() => ({ 
  count2: 0, 
  someFunction: () => {
    const count = useStoreApi.getState().count   // this line fires an error of invalid hook usage
    // ...
  },
}))
```
The above use always ends up with an error of invalid hook usage, It considers that hook can not be used outside of React component.

Is there any other way to call cross-store actions that are derived from `zustand/context` API? @dai-shi

**Reply by dai-shi** [reply]
No, that's technically not possible, because a store in context is not global (not just one).
Please consider creating just one combined store for your use case.

**Reply by Nishchit14** [reply]
Thank you for the quick response @dai-shi, zustand is amazing!!

**Reply by NamanRajputXXI** [reply]
how to access store data in another store in typescript store, Like i want to access one boolean state from another store.

**Reply by NamanRajputXXI** [reply]
@dai-shi

**Reply by dai-shi** [reply]
You can access with `useFooStore.getState()`. But, as it's not reactive, it might not be something you are looking for.

**Reply by NamanRajputXXI** [reply]
my store is look like this i want to split into 3 different store but want to access and isNavbarOpen in both 3 store and they should be reactive too  or can i store alll things in single (is this approach is good). @dai-shi 
import create from "zustand";
import { persist } from "zustand/middleware";
interface NavigationStore {
  isNavbarOpen: boolean;
  handleClickOnWelcome: () => void;
  isWelcomePanelActive: boolean;
  isToggled: boolean;
  handleClickOnNewButton: () => void;
  handleClickForToggle: () => void;
}
 const navigationUseStore = create(
  persist<NavigationStore>(
    (set) => ({
      isWelcomePanelActive: false, //for welcome component
      isNavbarOpen: false, //for navbar by default is closed
      handleClickOnWelcome: () => {
        set(() => ({
          isNavbarOpen: true,
          isWelcomePanelActive: true,
        }));
      },
      isToggled: false,
      handleClickOnNewButton: () => {
        set(() => ({
          isWelcomePanelActive: false,
          isNavbarOpen: false,
          isToggled: false,
        }));
      },
      handleClickForToggle: () => {
        set((state) => ({
          isToggled: !state.isToggled,
        }));
      },
    }),
    {
      name: "navigation-state",
      getStorage: () => localStorage,
    }
  )
);
export default navigationUseStore

**Reply by dai-shi** [reply]
> they should be reactive too or can i store alll things in single

Yeah, in such use case, a single store would be better.

**Reply by NamanRajputXXI** [reply]
ok thanks 👍 @dai-shi

**Comment by mghase** [comment]
You can use custom hook
[CUSTOM HOOK](https://stackoverflow.com/questions/74046364/zustand-get-state-from-another-zustand-store/75438333#75438333)

**Comment by amir-gorji** [comment]
Is it possible to do `useStore.getState().count` outside of a React component @dai-shi ?
I only need to know if it's possible to access an action in another store and do like `useStore.getState().actions.clear()` so the other store will be cleared.
Suppose I want to do that in another store that handles logging out.

**Reply by dbritto-dev** [reply]
Yes, you can. Here you go https://github.com/pmndrs/zustand#readingwriting-state-and-reacting-to-changes-outside-of-components
